modules/core/src/main/java/org/apache/synapse/config/xml/EnrichMediatorFactory.java

```diff
@@ -187,4 +200,49 @@ public QName getTagQName() {
         return XML_Q;
     }
 
+    private void validateTypeCombination(OMElement sourceEle, OMElement targetEle) {
+        int sourceType = -1;
+        int targetType = -1;
+
+        // source type attribute
+        OMAttribute sourceTypeAttr = sourceEle.getAttribute(ATT_TYPE);
+        if (sourceTypeAttr != null && sourceTypeAttr.getAttributeValue() != null) {
+            sourceType = convertTypeToInt(sourceTypeAttr.getAttributeValue());
+
+            // Source type is different form the existing (0-custom, 1-envelope, 2-body, 3-property, 4-inline)
+            if (sourceType < 0) {
+                logger.info("Un-expected source type");
+            }
+        }
+
+        // target type attribute
+        OMAttribute targetTypeAttr = targetEle.getAttribute(ATT_TYPE);
+        if (targetTypeAttr != null && targetTypeAttr.getAttributeValue() != null) {
+            targetType = convertTypeToInt(targetTypeAttr.getAttributeValue());
+
+            // check if target type is different form the existing (0-custom, 1-envelope, 2-body, 3-property, 4-inline)
+            if (targetType < 0) {
+                logger.info("Un-expected target type");
+            }
+            // check if target type is 4-inline
+            if (targetType == 4) {
+                throw new SynapseException("Inline not support for target attribute");
+            }
+        }
+        /*
+            check the wrong combination such as
+            sourceType = 1-envelope and targetType = 0-custom
+            sourceType = 1-envelope and targetType = 2-body
+            sourceType = 2-body and targetType = 2-body
+            sourceType = 0-custom and targetType = 1-envelope
```

Comment:
regarding sourceType = custom targetType = envelope and action=child

Can you please check, in above combination adding the soap header block using inline custom soap header block is possible in enrich mediator implementation before this PR ? If yes we have to consider this as a valid case.

Reply:
for the above combination 
when we have targetType = envelope, then the source value should be an envelope. Otherwise it should show the following error. "ERROR - EnrichMediator SOAPEnvelope is expected"
If we have targetType = envelope and an source value also envelope, then the output will be an envelope within an envelope. So this is not a valid combination.